### ISSUE TITLE
IllegalAccessException when using Actors with DexGuard

### ISSUE THREAD

**Issue by ZakTaccardi** [Kotlin/kotlinx.coroutines#319]
java.lang.IllegalAccessError: kotlinx.coroutines.experimental.channels.ActorCoroutine

DexGuard version: `8.1.13`
Kotlin Coroutines version: `0.22.5`
AGP version: `3.1.0`

A bit hard to get a better stack trace, but removing Actors and replacing them with `ArrayBroadcastChannel()` fixed the issue.

**Comment by elizarov** [comment]
Can you share your project or reproduce it with some self-contained code?

**Comment by ZakTaccardi** [comment]
just to note - no longer having this issue. I'm not sure if a newer version of coroutines fixed this issue or if it was a newer version of dexguard

**Comment by ZakTaccardi** [comment]
just kidding, was still having the issue! But the following proguard rules fixed it on both `0.26.1` and `0.30.2`

```
## Actors currently cause a VerifyError with DexGuard < 8.3
-keep,allowobfuscation,includecode class kotlinx.coroutines.experimental.channels.ActorCoroutine { *; }
```
